src/hex.rs

```diff
@@ -84,6 +97,40 @@ impl<'a> Iterator for HexIterator<'a> {
             Some(Ok(ret as u8))
         }
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let len = self.sl.len() / 2;
+        (len, Some(len))
+    }
+}
+
+impl<'a> DoubleEndedIterator for HexIterator<'a> {
+    fn next_back(&mut self) -> Option<Result<u8, Error>> {
+        let current_len = self.sl.len();
+        if current_len % 2 == 1 {
+            Some(Err(Error::OddLengthString(self.sl.len())))
+        } else if self.sl.is_empty() {
```

Comment:
Why not use `current_len` here as well? I would start with `if current_len == 0 { ... } else if current_len % 2 == 1 { ... }` personally.

Reply:
But I see this is mirroring the impl of `Iterator`, which if fine I guess.